ruby/engine/embedded_help.rb

```diff
@@ -637,45 +639,61 @@ def self.glob(pattern, *args, **options)
       pattern_array = pattern
     end
 
-    #puts "Dir.glob pattern = #{pattern}, pattern_array = #{pattern_array}, args = #{args}, options = #{options}"
+    pattern_has_embedded = pattern_array.any? {|p| p.to_s.chars.first == ':'}
+    base_has_embedded = (!base.nil? && base.to_s.chars.first == ':')
+    if !pattern_has_embedded && !base_has_embedded
+      # puts "Original glob"
+      return self.original_glob(pattern, flags: flags, base: base, sort: sort)
+    end
+
+    absolute_base = if base.nil?
+                      nil
+                    else
+                      OpenStudio.get_absolute_path(base)
+                    end
```

Comment:
If the base is provided, get_absolute_path it, which will just do File.expand_path base if base is not embedded. I could have used File.expand_path or File.absolute_path I guess, which we override.